homeassistant/components/icloud/account.py

```diff
@@ -173,11 +174,12 @@ def update_devices(self) -> None:
             _LOGGER.error("Unknown iCloud error: %s", err)
             self._fetch_interval = 2
             dispatcher_send(self.hass, self.signal_device_update)
-            track_point_in_utc_time(
-                self.hass,
-                self.keep_alive,
-                utcnow() + timedelta(minutes=self._fetch_interval),
-            )
+            if not self._config_entry.pref_disable_polling:
+                track_point_in_utc_time(
+                    self.hass,
+                    self.keep_alive,
+                    utcnow() + timedelta(minutes=self._fetch_interval),
+                )
```

Comment:
Since there is duplicate coded below, this should be moved into a new function.

It also looks like there is a bug here that the callable that `track_point_in_utc_time` returns is never saved or used to cancel the fetch if the integration is unloaded before it can fire

Reply:
Does 1 line of code really count as duplicated code though?

I thought about doing it but then looked slightly uglier as the new function still needs to take some parameters.

Related to the return value, probably better to leave it for a different MR, as I'm not confortable changing something I didn't introduce nor detect 😀

Reply:
> Does 1 line of code really count as duplicated code though?

Yes
> 
> I thought about doing it but then looked slightly uglier as the new function still needs to take some parameters.
> 
> Related to the return value, probably better to leave it for a different MR, as I'm not confortable changing something I didn't introduce nor detect 😀

The bug should be fixed in a separate PR 👍

Reply:
✅